### DISCUSSION TITLE
有没有办法获取一个函数的entry块

### DISCUSSION THREAD

**Discussion by Aurora-1412** [arcsysu/SYsU-lang#52] [Q&A] [answered]
除了在所有函数中都加一个entry块参数之外，还有没有其他的办法呢。
例如我已知当前块是BB，如何获得BB前面/后面的块呢
看到一个API是getBasicBlockList，但我调用auto entryBB = TheFunction->getBasicBlockList().front();时，会提示（已隐式声明）--它是已删除的函数 ，没找到解决办法。

**Comment by Thirtory** [comment, marked as answer]
BB->getPrevNode(); BB->getNextNode(); //得到前后 当然有时候会是空
auto entryBB = &TheFunction->getBasicBlockList().front(); // 可能是不允许拷贝构造吧 用指针指向它就行应该 不是很确定

**Comment by Aurora-1412** [comment]
谢谢～



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年5月24日(周二) 下午4:59
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [arcsysu/SYsU-lang] 有没有办法获取一个函数的entry块 (Discussion #52)




 
BB-&gt;getPrevNode(); BB-&gt;getNextNode(); //得到前后 当然有时候会是空
 auto entryBB = &amp;TheFunction-&gt;getBasicBlockList().front(); // 可能是不允许拷贝构造吧 用指针指向它就行应该 不是很确定
